### PR TITLE
spring-boot-loader Performance and other optimizations backports from 2.1.x

### PR DESCRIPTION
**Pull request by l0c0f1x**
Hello,

by closing gh-12226, gh-12296 and gh-12986 some optimizations and fixes regarding the performance of loading JAR files were introduced. These changes were only introduced in the 2.1.x codebase and are written for JDK >=1.8 compatibility.
This PR backports these changes to the 1.5.x codebase and restores JDK >=1.6 compatibility.

Background:
Currently we are migrating a legacy Java1.7 RichClient codebase to Maven and are using the spring-boot-maven plugin to produce a FatJar. However, because of some constraints we need to keep ~30 JAR's external to the produced FatJar and use the `PropertiesLauncher` to load these.
Using v1.5.22.RELEASE the startup of the application takes ~30 sec. With the changes from this PR startup takes ~3 sec.

Let me know if anything is wrong.

Thanks.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@l0c0f1x Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=18515)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-boot&pullRequestId=18515) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@l0c0f1x Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=18515)!

File: spring-boot-tools/spring-boot-loader/src/main/java/org/springframework/boot/loader/jar/CentralDirectoryFileHeader.java [hunk 1]
```diff
@@ -118,7 +118,7 @@ public long getTime() {
 
 	/**
 	 * Decode MS-DOS Date Time details. See
-	 * <a href="http://mindprod.com/jgloss/zip.html">mindprod.com/jgloss/zip.html</a> for
+	 * <a href="https://mindprod.com/jgloss/zip.html">mindprod.com/jgloss/zip.html</a> for
```
Comment (l0c0f1x) [inline spring-boot-tools/spring-boot-loader/src/main/java/org/springframework/boot/loader/jar/CentralDirectoryFileHeader.java]: This page is actually only reachable via `http` but spring-javaformat checkstyle requires every Link to be `https`. Is there something to do about this? Maybe adding this info in the comment, too?

**Comment by PascalSchumacher** [comment]
Spring Boot 1.5 is not maintained (see: https://spring.io/blog/2019/08/06/it-is-time-goodbye-spring-boot-1-x) any more so I fear you are wasting your time.

**Comment by l0c0f1x** [comment]
> 
> 
> Spring Boot 1.5 is not maintained any more (see: https://spring.io/blog/2019/08/06/it-is-time-goodbye-spring-boot-1-x), so I fear you are wasting your time.

Ok, thanks for the info, didn't catch that. Guess we will be using this in a custom build for us then until we also migrate to a new JDK version compatible with Spring-Boot 2.
Closing this PR.

**Comment by wilkinsona** [comment]
Thanks, @PascalSchumacher. That's correct.

**Pull request closed without merging**